src/OrleansRuntime/LogConsistency/ProtocolServices.cs

```diff
@@ -21,45 +19,61 @@ namespace Orleans.Runtime.LogConsistency
     /// </summary>
     internal class ProtocolServices : ILogConsistencyProtocolServices
     {
-
-        public GrainReference GrainReference { get { return grain.GrainReference; } }
-
-        private Logger log;
-        private readonly IInternalGrainFactory grainFactory;
-
-        public IMultiClusterRegistrationStrategy RegistrationStrategy { get; private set; }
-
-        private Grain grain;   // links to the grain that owns this service object
+        private static readonly object[] EmptyObjectArray = new object[0];
 
         // pseudo-configuration to use if there is no actual multicluster network
-        private static MultiClusterConfiguration PseudoMultiClusterConfiguration;
+        private static readonly Interner<string, MultiClusterConfiguration> PseudoMultiClusterConfigurations = new Interner<string, MultiClusterConfiguration>();
```

Comment:
why do you need an interner? if used, the PseudoMultiClusterConfiguration will be a single one... just a config with the current clusterId... this is not multi-valued. Or am I misunderstanding its use?